libopenage/unit/attribute.h

```diff
@@ -167,14 +169,44 @@ template<> class Attribute<attr_type::hitpoints>: public AttributeContainer {
 	float hp_bar_height;
 };
 
+template<> class Attribute<attr_type::armor>: public AttributeContainer {
+public:
+	Attribute(typeamount_set a)
+		:
+		AttributeContainer{attr_type::armor},
+		armor{a} {}
+
+	bool shared() const override {
+		return true;
+	}
+
+	std::shared_ptr<AttributeContainer> copy() const override {
+		return std::make_shared<Attribute<attr_type::armor>>(*this);
+	}
+
+	typeamount_set armor;
+};
+
 template<> class Attribute<attr_type::attack>: public AttributeContainer {
 public:
+	// TODO remove (keep for testing)
 	Attribute(UnitType *type, coord::phys_t r, coord::phys_t h, unsigned int d, UnitType *reset_type)
 		:
 		AttributeContainer{attr_type::attack},
 		ptype{type},
 		range{r},
 		init_height{h},
+		stance{attack_stance::do_nothing},
+		attack_type{reset_type} {
+		damage[4] = d;
```

Comment:
what is index 4? if it's a special value that could be named, please use an `enum class`